### ISSUE TITLE
Docker supabase-db container fails to start

### ISSUE THREAD

**Issue by shahradelahi** [supabase/supabase#12683]
# Bug report

```
2023-02-26 11:42:49 supabase-db | The database cluster will be initialized with this locale configuration:
2023-02-26 11:42:49 supabase-db |   provider:    libc
2023-02-26 11:42:49 supabase-db |   LC_COLLATE:  en_US.utf8
2023-02-26 11:42:49 supabase-db |   LC_CTYPE:    C.UTF-8
2023-02-26 11:42:49 supabase-db |   LC_MESSAGES: en_US.utf8
2023-02-26 11:42:49 supabase-db |   LC_MONETARY: en_US.utf8
2023-02-26 11:42:49 supabase-db |   LC_NUMERIC:  en_US.utf8
2023-02-26 11:42:49 supabase-db |   LC_TIME:     en_US.utf8
2023-02-26 11:42:49 supabase-db | The default database encoding has accordingly been set to "UTF8".
2023-02-26 11:42:49 supabase-db | The default text search configuration will be set to "english".
2023-02-26 11:42:49 supabase-db | 
2023-02-26 11:42:49 supabase-db | Data page checksums are disabled.
2023-02-26 11:42:49 supabase-db | 
2023-02-26 11:42:49 supabase-db | fixing permissions on existing directory /var/lib/postgresql/data ... ok
2023-02-26 11:42:49 supabase-db | creating subdirectories ... ok
2023-02-26 11:42:49 supabase-db | selecting dynamic shared memory implementation ... posix
2023-02-26 11:42:49 supabase-db | selecting default max_connections ... 20
2023-02-26 11:42:50 supabase-db | selecting default shared_buffers ... 400kB
2023-02-26 11:42:50 supabase-db | selecting default time zone ... Etc/UTC
2023-02-26 11:42:50 supabase-db | creating configuration files ... ok
2023-02-26 11:42:50 supabase-db | 2023-02-26 08:12:50.098 UTC [83] FATAL:  data directory "/var/lib/postgresql/data" has invalid permissions
2023-02-26 11:42:50 supabase-db | 2023-02-26 08:12:50.098 UTC [83] DETAIL:  Permissions should be u=rwx (0700) or u=rwx,g=rx (0750).
2023-02-26 11:42:50 supabase-db | child process exited with exit code 1
2023-02-26 11:42:50 supabase-db | initdb: removing contents of data directory "/var/lib/postgresql/data"
2023-02-26 11:42:50 supabase-db | running bootstrap script ... 2023-02-26T08:12:50.903389200Z The files belonging to this database system will be owned by user "postgres"
```

## Describe the bug

I Wanted to setup supabase locally using docker. but the supabase-db container fails to start.

It's clear that the permissions are not set correctly. 

I tried to set the permissions manually but i don't know how because the container is not running.

## To Reproduce

Follow the steps in the [documentation](https://supabase.com/docs/guides/self-hosting/docker) to setup the project.

## Expected behavior

The supabase-db container should start.

## System information

- `OS`: `Windows`
- `DOCKER`: `Docker version 20.10.22, build 3a2c30b`

**Comment by supermar1010** [comment]
That's probalby a problem with using docker for windows. I use this on linux and it works flawlessly. Maybe you can search for "postgresql docker windows permission problem" or something. I doubt this is a supabase problem. As they are just using the official postgresql image and are mapping some data in via volumes

**Comment by shahradelahi** [comment]
@supermar1010 Thanks for your reply. I tried to run it on Linux and it works fine. For the Windows version, I switch the docker engine from `Hyper-V` to `WSL2`, and it also works fine. I don't know why, but the problem was with `Hyper-V` engine.

**Comment by sweatybridge** [comment]
Thanks for posting the workaround. I will close this as resolved then.

**Comment by mjolnirPH** [comment]
> @supermar1010 Thanks for your reply. I tried to run it on Linux and it works fine. For the Windows version, I switch the docker engine from `Hyper-V` to `WSL2`, and it also works fine. I don't know why, but the problem was with `Hyper-V` engine.

Pardon for reopening this resolved issue (for @sweatybridge), but I'm running my Docker Windows using the WSL2 engine, and I have the same problem. 

I'd appreciate it very much of you could help me find a solution.

Thanks.

**Comment by grquin** [comment]
@mjolnirPH 

> > @supermar1010 Thanks for your reply. I tried to run it on Linux and it works fine. For the Windows version, I switch the docker engine from `Hyper-V` to `WSL2`, and it also works fine. I don't know why, but the problem was with `Hyper-V` engine.
> 
> Pardon for reopening this resolved issue (for @sweatybridge), but I'm running my Docker Windows using the WSL2 engine, and I have the same problem.
> 
> I'd appreciate it very much of you could help me find a solution.
> 
> Thanks.

Same here. WSL2 is enabled for the version of Ubuntu I'm running, but in the supabase container, postgres continues to be stuck on "restarting".

**Comment by grquin** [comment]
I fixed the issue. When running a docker container via WSL2, there is a permissions issue with postgresql directory path for volumes. I then created a supabase_data directory outsideo of the docker container for the volume and edited the docker-compose.yml on line 379 from this:

` # PGDATA directory is persisted between restarts
      - ./volumes/db/data:/var/lib/postgresql/data:Z` 
      
  To this:
  
  `# PGDATA directory is persisted between restarts
      - /mnt/c/Repos/supabase_data:/mnt/c/Repos/supabase_data:Z`
      
      
    Ran: 
 $ docker-compose up -d
[+] Running 12/12
 ✔ Container supabase-imgproxy               Started                                                               0.0s
 ✔ Container supabase-vector                 Healthy                                                               0.0s
 ✔ Container supabase-db                     Healthy                                                               0.1s
 ✔ Container supabase-analytics              Healthy                                                               0.1s
 ✔ Container realtime-dev.supabase-realtime  Started                                                               0.2s
 ✔ Container supabase-kong                   Started                                                               0.2s
 ✔ Container supabase-auth                   Started                                                               0.2s
 ✔ Container supabase-studio                 Started                                                               0.2s
 ✔ Container supabase-meta                   Started                                                               0.2s
 ✔ Container supabase-rest                   Started                                                               0.2s
 ✔ Container supabase-edge-functions         Started                                                               0.2s
 ✔ Container supabase-storage                Started
 
 Back up and running.
